chainer/function_set.py

```diff
@@ -28,8 +31,20 @@ def __init__(self, **functions):
                 object as attributes.
 
         """
-        for name, func in six.iteritems(functions):
-            setattr(self, name, func)
+
+        for key, func in six.iteritems(functions):
+            if key == 'name':
```

Comment:
This validation affects to existing user who uses ``name`` as the name of a layer.